packages/flutter/lib/src/material/button_bar.dart

```diff
@@ -26,43 +33,117 @@ import 'raised_button.dart';
 ///  * [FlatButton], another kind of button.
 ///  * [Card], at the bottom of which it is common to place a [ButtonBar].
 ///  * [Dialog], which uses a [ButtonBar] for its actions.
-///  * [ButtonTheme], which configures the [ButtonBar].
+///  * [ButtonBarTheme], which configures the [ButtonBar].
 class ButtonBar extends StatelessWidget {
   /// Creates a button bar.
   ///
-  /// The alignment argument defaults to [MainAxisAlignment.end].
+  /// Both [buttonMinWidth] and [buttonHeight] must be non-negative if they
+  /// are not `null`.
   const ButtonBar({
     Key key,
-    this.alignment = MainAxisAlignment.end,
-    this.mainAxisSize = MainAxisSize.max,
+    this.alignment,
+    this.mainAxisSize,
+    this.buttonTextTheme,
+    this.buttonMinWidth,
+    this.buttonHeight,
+    this.buttonPadding,
+    this.buttonAlignedDropdown,
+    this.layoutBehavior,
     this.children = const <Widget>[],
-  }) : super(key: key);
+  }) : assert(buttonMinWidth == null || buttonMinWidth >= 0.0),
+       assert(buttonHeight == null || buttonHeight >= 0.0),
+       super(key: key);
 
   /// How the children should be placed along the horizontal axis.
+  ///
+  /// If `null` then it will use [ButtonBarTheme.alignment]. If that is `null`,
+  /// it will default to [MainAxisAlignment.end].
   final MainAxisAlignment alignment;
 
   /// How much horizontal space is available. See [Row.mainAxisSize].
+  ///
+  /// If `null` then it will use the surrounding [ButtonBarTheme.mainAxisSize].
+  /// If that is `null`, it will default to [MainAxisSize.max].
   final MainAxisSize mainAxisSize;
 
+  /// Overrides the surrounding [ButtonTheme.textTheme] to define a button's
+  /// base colors, size, internal padding and shape.
+  ///
+  /// If `null` then it will use the surrounding [ButtonBarTheme.buttonTextTheme].
+  /// If that is `null`, it will default to [ButtonTextTheme.primary].
+  final ButtonTextTheme buttonTextTheme;
+
+  /// Overrides the surrounding [ButtonThemeData.minWidth] to define a button's
+  /// minimum width.
+  ///
+  /// If `null` then it will use the surrounding [ButtonBarTheme.buttonMinWidth].
+  /// If that is `null`, it will default to `64.0` logical pixels.
```

Comment:
We don't usually put numbers in back ticks either. Or true and false (see below).

Reply:
Ok, I thought it was useful to call out constants in the descriptions, but I will take these out.